src/components/StatesButton/StatesButton.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import style from './StatesButton.st.css';
+import { Button, SIZE } from '../Button';
+import { Check } from '../../icons/dist';
+
+const delay = time => new Promise(resolve => setTimeout(resolve, time));
+
+export interface StatesButtonProps {
+  className?: string;
+  disabled: boolean;
+  onClick: Function;
+  text: string;
+}
+
+interface StatesButtonState {
+  success: boolean;
+}
+
+export class StatesButton extends React.Component<
```

Comment:
We'll need to find a different name. Maybe `SuccessButton`.
I'll get back to you with this tomorrow

Reply:
@shlomitc suggested `StatesButton` so please decide with him what name you want
(`ProgressButton` sounds better imo)

Reply:
@shlomitc  what do you think?

Reply:
I got @shlomitc's approval for the name 😄